server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -294,7 +254,11 @@ public FireDepartmentMetrics getMetrics()
     @Override
     public void run()
     {
-      plumber = initPlumber();
+      Firehose firehose = null;
+      FirehoseV2 firehoseV2 = null;
+      boolean normalExit = true;
```

Comment:
Saner to set `normalExit = false` initially, and set it to `true` in the end of try block. Otherwise `Throwable` which is not a `RuntimeException` in try block is qualified as "normal exit".